types/react/index.d.ts

```diff
@@ -79,7 +79,7 @@ declare namespace React {
      * @deprecated Please use `ElementType`
      */
     type ReactType<P = any> = ElementType<P>;
-    type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
+    type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P> | VoidFunctionComponent<P>;
```

Comment:
This might cause CI to fail? Or is `master` currently broken?

Reply:
`master` seems to be currently broken

Reply:
Thanks, this does seem to be the problem. I'm trying it out now with just using VoidFunctionComponent here. Does this look okay to you?

```suggestion
    type ComponentType<P = {}> = ComponentClass<P> | VoidFunctionComponent<P>;
```

Reply:
Ah, this is no good either. Going to try by reverting this line without `VoidFunctionComponent`.